datafusion/common/src/config.rs

```diff
@@ -454,6 +470,80 @@ config_namespace! {
     }
 }
 
+#[cfg(feature = "parquet")]
+impl ParquetOptions {
+    /// Convert the global session options, [`ParquetOptions`], into a single write action's [`WriterPropertiesBuilder`].
+    ///
+    /// The returned [`WriterPropertiesBuilder`] can then be further modified with additional options
+    /// applied per column; a customization which is not applicable for [`ParquetOptions`].
+    pub fn writer_props_from_global_opts(&self) -> Result<WriterPropertiesBuilder> {
```

Comment:
I think this is basically the API described in https://github.com/apache/datafusion/issues/11367 (cc @Rachelint )